src/shortcuts/newThoughtAbove.js

```diff
@@ -11,6 +12,11 @@ const Icon = ({ fill = 'black', size = 20, style }) => <svg version="1.1" classN
   </g>
 </svg>
 
+const exec = () => {
+  return (dispatch, getState) => {
+    newThought({ insertBefore: true, thunkDispatch: dispatch, getState })
+  }
+}
```

Comment:
Let's invert this by making `newThought` itself an `action-creator` that returns a thunk. `newThought` is used in several places, but they are all shortcuts that should be dispatching thunk actions anyway. (I am slowly picking out the `util` functions that should actually be `action-creator` functions.)

I suggest leaving this as-is in the current PR and making the change in your next PR for converting shortcut execs to thunks.

Reply:
Sure will do that